src/kernels/ipywidgets-message-coordination/notebookIPyWidgetCoordinator.ts

```diff
@@ -210,63 +201,5 @@ export class NotebookIPyWidgetCoordinator implements IExtensionSyncActivationSer
         const coordinator = this.messageCoordinators.get(notebook);
         void coordinator?.then((c) => c.dispose());
         this.messageCoordinators.delete(notebook);
-
-        this.attachedEditors.delete(notebook);
-    }
-    private attachCoordinator(
```

Comment:
I combined this code and the 'initialize' function for the CommonMessageCoordinator into a single function. Makes this code a lot simpler and alleviates the problem with reopening.

On reopen this series of events would occur:

1. Doc Open = creates coordinator and sticks a promise for initialize into a map
2. Kernel selection change event = deletes promise from map
3. Promise finishes anyway and attaches the coordinator by putting it into the 'attachedEditors' map. This may or may not happen before the kernel selection event.
4. Selection change creates a new coordinator but then when it attaches, the 'attachedEditors' says we already have one so it doesn't attach the new coordinator
5. Coordinator doesn't send the events.

Change here was to just have one map tracking the coordinators - messageCoordinators. Then we don't update a map during a promise and it always attaches.